hack/mock-based-tests.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+openshift admin new-project mock --description="This is an example project to demonstrate OpenShift v3" --admin="e2e-user"
+
+osc project mock
+
+# upload the template
+osc create -f examples/mock-app-template/mock-template.json -n mock
+
+# create the config
+
+# create the new app
+osc new-app mock-app-sample -p MOCK_NAME=mock1
+
+# wait for
+echo "[INFO] Waiting pod mock"
+wait_for_command "osc get -n mock pods -l name=mock-app-pod | grep -i Running" $((60*TIME_SEC))
```

Comment:
Those tests are not intended to be included "as is" this PR is here to show how to use the openshift/mock-app image. This was asked from another parent PR https://github.com/openshift/origin/pull/1903, mock-app is a tool to easily add non regression tests. 

BTW those tests are not long, no computation, only openshift/kubernetes work (services start, pods starts , rc resizes then assertions)